test-app-reactive/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/integration/TelemetryPropagationTest.kt

```diff
@@ -56,6 +58,49 @@ class TelemetryPropagationTest : IntegrationTestBase() {
     )
   }
 
+  @Test
+  fun `telemetry information is propagated between publishers and listeners for queues`() = runTest {
+    // Given a span
+    val span = withSpan {
+      // When I publish an OFFENDER_MOVEMENT-RECEPTION message
+      val event = HmppsEvent("event-id", "OFFENDER_MOVEMENT-RECEPTION", "some event contents")
+      inboundSqsOnlyClient.sendMessage(
+        SendMessageRequest.builder()
+          .queueUrl(inboundSqsOnlyQueueUrl)
+          .messageBody(gsonString(event))
+          .messageAttributes(
+            mapOf(
+              "eventType" to MessageAttributeValue.builder().dataType("String").stringValue(event.type).build(),
+            ),
+          )
+          .build(),
+      )
+    }
+
+    // And the OFFENDER_MOVEMENT-RECEPTION message is consumed, resulting in an offender.movement.reception message being published
+    await untilCallTo {
+      outboundSqsOnlyTestSqsClient.countMessagesOnQueue(outboundSqsOnlyTestQueueUrl).get()
```

Comment:
slightly annoying, but in order to duplicate the test I had to create two emitters.  This wasn't necessary before as we had two queues subscribed to the same topic.  Since we are publishing straight to queues I therefore had to add in an additional publish at the end so that the message would then appear on the outbound test queue.